### ISSUE TITLE
function as constructor to new not compiled to arrow style

### ISSUE THREAD

**Issue by alessbelli** [decaffeinate/decaffeinate#1385]
Decaffeinate is producing the wrong JavaScript based on my CoffeeScript input:

```coffee
a =  new Promise (resolve) -> resolve()
```
 [(repl)](https://decaffeinate-project.org/repl/#?useCS2=false&useJSModules=false&loose=false&evaluate=true&stage=full&code=a%20%3D%20%20new%20Promise%20(resolve)%20-%3E%20resolve())

I get this output:

```js
const a =  new Promise(function(resolve) { return resolve(); });
```

Here's what I expect it to be instead:

```js
const  a = new Promise(resolve => resolve());
```

**Comment by eventualbuddha** [comment]
This is happening because esnext (which is in the process of being renamed to resugar) is too broad about how it disallows transforming functions into arrow functions. Specifically, it prevents the transformation [if the parent is a `NewExpression`](https://github.com/resugar/resugar/blob/09fd77f8879c6d9c6fd3d84a11875630eea7a61d/packages/resugar-codemod-functions-arrow/src/index.ts#L34-L36), but that should only happen if the parent is a `NewExpression` *and* the function is the parent's `callee`. This is a fairly easy fix if you want to create a PR for it.

**Comment by eventualbuddha** [comment]
This is now fixed in v6.0.2.
